packages/headless/src/app/product-recommendation-engine/product-recommendation-engine.ts

```diff
@@ -74,7 +74,7 @@ export function buildProductRecommendationEngine(
 
   const thunkArguments = {
     ...buildThunkExtraArguments(options.configuration, logger),
-    searchAPIClient,
+    apiClient: searchAPIClient,
```

Comment:
Hmm, I'm curious to know why didn't typescript catch this? Maybe we can add `SearchThunkExtraArguments` on the `thunkArguments` variable?

Maybe something more we could do is create a `buildSearchThunkExtraArguments` function and reuse it to cut the duplication inside `search-engine.ts`?

I also expect the same error should be happening with the recommendation engine, so we'd need to update that place too.

Last thing, we should keep both keys `searchAPIClient` and `apiClient` while Headless remains on v1 just in case. With v2, we will be free to break and remove `searchAPIClient`.

Reply:
I added the type `SearchThunkExtraArgument` and the legacy key `searchAPIClient`